### PR TITLE
WIP: Integrate astyle into the build

### LINKED ISSUE #3071: Integrate a code formatter into the build
A code formatter, mainly for C++, should be integrated into the build. The formatter should follow the [coding conventions](https://github.com/LMMS/lmms/wiki/Coding-conventions), and the configuration file should become the reference. It should be able to format files (for developers modifying code) and check whether files follow the format (for the build process). Only a subset should be formatted, excluding third-party files.

Existing solutions (`astyle`, `bcpp`, `indent`, `trueprint`) should be explored before developing a custom tool.

### PR DESCRIPTION
**Pull request by zapashcanon**
Hi,

I did this to fix #3071.

There are 3 commits.

The first one add astyle configuration files following LMMS coding conventions guidelines. I'm not sure about a few options so I put them at the end of `.astylerc` and commented them, but, I'll be glad to hear about other options too. Once every one is happy with one solution, I'll rebase and update this commit. If you don't know astyle, here is [the doc](http://astyle.sourceforge.net/astyle.html) with a lot of examples.

The second commit is just a `.gitignore` update to ignore `.orig` files, `.orig` files are backup made by astyle. It's also possible to disable backups but I'd rather disable it as default, so, this `.gitignore` update seemed legit to me.

The last one is just a test using:

```sh
astyle --options=./.astylerc ./*.cpp ./*.hpp ./*.c ./*.h ./*.C
```

Thus everyone can see how the code look using astyle. I don't know if I should keep this commit or not, so just tell me. It's also to see if buil will break or not, it shouldn't, be as it affects a lot of files...

I'm also not sure if I should add the above command in a script somewhere or even in the `.travis.yml` and optionally make the build fail if some errors are present... what do you think ?

### CODE REVIEW

File: buildtools/bin2res.cpp [hunk 1]
```diff
@@ -39,124 +39,131 @@ struct embed
...
 		std::cout << std::endl << "};" << std::endl << std::endl;
 		delete[] data;
 	}
 
-	if( embedded_data.size() > 0 )
+	if ( embedded_data.size() > 0 )
 	{
 		std::cout << "static const unsigned char dummy_data[] ="
-					"{ 0x00 };" << std::endl << std::endl;
+			  "{ 0x00 };" << std::endl << std::endl;
```
Comment (Umcaruje) [inline buildtools/bin2res.cpp]: why are these tabs getting converted to spaces?

File: include/AudioAlsa.h [hunk 2]
```diff
@@ -66,13 +66,13 @@ class AudioAlsa : public AudioDevice, public QThread
...
 public:
-	AudioAlsa( bool & _success_ful, Mixer* mixer );
+	AudioAlsa( bool & _success_ful, Mixer * mixer );
 	virtual ~AudioAlsa();
 
 	inline static QString name()
 	{
 		return QT_TRANSLATE_NOOP( "setupWidget",
-			"ALSA (Advanced Linux Sound Architecture)" );
+					  "ALSA (Advanced Linux Sound Architecture)" );
```
Comment (zapashcanon) [inline include/AudioAlsa.h]: Here.

File: include/AudioDevice.h [hunk 3]
```diff
@@ -105,28 +105,28 @@ class AudioDevice
 	// convert a given audio-buffer to a buffer in signed 16-bit samples
 	// returns num of bytes in outbuf
 	int convertToS16( const surroundSampleFrame * _ab,
-						const fpp_t _frames,
-						const float _master_gain,
-						int_sample_t * _output_buffer,
-						const bool _convert_endian = false );
+			  const fpp_t _frames,
+			  const float _master_gain,
+			  int_sample_t * _output_buffer,
```
Comment (zapashcanon) [inline include/AudioDevice.h]: And here.

File: buildtools/bin2res.cpp (hunk 1 above)
Comment (zapashcanon) [inline buildtools/bin2res.cpp]: I tried a few things, seems like astyle always try to do the best visual alignement and that it's not possible to get completely rid of spaces. I managed to get it doing it a little better, but there's still some spaces, see following commented code.

**Pull request closed without merging**